go/vt/sqlparser/parse_test.go

```diff
@@ -3485,6 +3485,37 @@ func TestCreateTableEscaped(t *testing.T) {
 	}
 }
 
+func TestCreateTableSelect(t *testing.T) {
+	testCases := []struct {
+		input  string
+		output string
+	}{{
+		input: "create table `t` as select * from `uv`",
+		output: "create table t as select * from uv",
+	}, {
+		input: "create table `t` as select * from `uv` LIMIT 5",
+		output: "create table t as select * from uv limit 5",
+	}, {
+		input: "create table `t` select pk from `foo`",
+		output: "create table t select pk from foo",
+	}, {
+		input: "create table t (pk int) select val from foo",
+		output: "create table t (\n" +
```

Comment:
Add some tests declaring indexes, multi-column primary keys, and adding options (like the `ENGINE` option)